### ISSUE TITLE
Patcher not following required subgroup rules?

### ISSUE THREAD

**Issue by leonardovcr** [Synthesis-Collective/SynthEBD#104]
Hey Piranha, just to let you know,  while looking through some logs I've noticed by chance that some required subgroup rules aren't being adhered to...

On the Leyenda Config Files, I have set rules for Hand Normals HaN.VigEx and HaN.Vig to only match Body Normals with VigEx and Vig suffixes, respectively. On many entries patcher seems to not respect that. 

Here are some log excerpts:

HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.Leaf|BN.Crvy.Norm|BS|BSP.Slk|HaD.Mol|HaN.VigEx|HaS|HaSP.Def
HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.Leaf|BN.Crvy.MuscLow|BS|BSP.Slk|HaD.Mol|HaN.VigEx|HaS|HaSP.Def
HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.Muff|BN.Crvy.MuscLow|BS|BSP.Gb|HaD.Mol|HaN.VigEx|HaS|HaSP.Def
HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.Muff|BN.Crvy.Norm|BS|BSP.Slk|HaD.Mol|HaN.VigEx|HaS|HaSP.Def
HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.MuffF|BN.Crvy.MuscHi|BS|BSP.Gb|HaD.Mol|HaN.Vig|HaS|HaSP.Def
HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.MuffF|BN.Slim.Norm|BS|BSP.Gb|HaD.Mol|HaN.Vig|HaS|HaSP.Def

Here an entry from the same log file where assignment is correct (BN.Slim.VigEx paired with HaN.VigEx):

HD.Def.DefM|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Mol.PN.MuffF|BN.Slim.VigEx|BS|BSP.Gb|HaD.Mol|HaN.VigEx|HaS|HaSP.Def

What is most curious, is that I didn't find a single combination where a VigEx/Vig Body Normal was paired with a wrong Hand Normal. They were all paired with their respective VigEx/Vig Hand Normals. There are however, many entries of VigEx/Vig Hand Normals being paired with all kinds of Body Normals. At first I thought it could be a cache issue, but behaviour is the same with cacheing enabled or disabled.

Maybe it's an issue with how I've set up those rules? I've set requirements both ways, as in VigEx Body Normal requiring VigEx Hand Normal, and VigEx Hand Normal requiring VigEx Body Normal for example.

Some pics:

![image](https://user-images.githubusercontent.com/13605450/221690843-da25fb90-29aa-4474-a832-754323f7503f.png)
![image](https://user-images.githubusercontent.com/13605450/221690947-5f2b78ae-db01-4e2c-8675-b182136cbd78.png)

And here's the config file: https://www.nexusmods.com/skyrimspecialedition/mods/73717?tab=files&file_id=362660

**Comment by Piranha91** [comment]
If you have a chance, can you please upload a verbose NPC log where this occurs so I can see what the patcher is thinking?

**Comment by leonardovcr** [comment]
There you go, looks like after filtering for required subgroups of Hand Normal, it doesn't backtrace to choose another Body Normal 🤔 

```
				Chose seed subgroup BD.Flat.Def.BN.LStripM in Leyenda Skin 4.0 at random
				<Seed-BD_Flat_Def_BN_LStripM>Available Subgroups:
					0: [HD.Adr.Ad,HD.Adr.AdM,HD.Alt.Alt,HD.Alt.AltM,HD.Asian.Asi,HD.Asian.AsiM,HD.Def.Def,HD.Def.DefM]
					1: [HN.Ad,HN.Alt,HN.As,HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Current Combination:
					
					Available Subgroups:
						0: [HD.Adr.Ad,HD.Adr.AdM,HD.Alt.Alt,HD.Alt.AltM,HD.Asian.Asi,HD.Asian.AsiM,HD.Def.Def,HD.Def.DefM]
					1: [HN.Ad,HN.Alt,HN.As,HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HD.Def.Def at position 0 at random.
					
					Trimming remaining subgroups within Leyenda Skin 4.0 by the required/excluded subgroups of HD.Def.Def
					
					Current Combination: HD.Def.Def
					
					Available Subgroups:
					0: [HD.Adr.Ad,HD.Adr.AdM,HD.Alt.Alt,HD.Alt.AltM,HD.Asian.Asi,HD.Asian.AsiM,HD.Def.Def,HD.Def.DefM]
						1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HN.Def at position 1 at random.
					
					Trimming remaining subgroups within Leyenda Skin 4.0 by the required/excluded subgroups of HN.Def
					
					Current Combination: HD.Def.Def , HN.Def
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
						2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HS at position 2 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
						3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HSP.Def at position 3 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
						4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HDE.BDM at position 4 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
						5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: BD.Flat.Def.BN.LStripM at position 5 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
						6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: BN.Crvy.Del at position 6 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM , BN.Crvy.Del
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
						7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: BS at position 7 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM , BN.Crvy.Del , BS
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
						8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: BSP.Slk at position 8 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM , BN.Crvy.Del , BS , BSP.Slk
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
						9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HaD.Def at position 9 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM , BN.Crvy.Del , BS , BSP.Slk , HaD.Def
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Del,BN.Crvy.Norm,BN.Crvy.Smo,BN.Crvy.Fat,BN.Crvy.MuscNo,BN.Crvy.MuscLes,BN.Crvy.MuscLow,BN.Crvy.MuscMed,BN.Crvy.MuscHi,BN.Crvy.MuscEx,BN.Crvy.Vig,BN.Crvy.VigEx,BN.Flat.Del,BN.Flat.Norm,BN.Flat.Smo,BN.Flat.Fat,BN.Flat.MuscNo,BN.Flat.MuscLes,BN.Flat.MuscLow,BN.Flat.MuscMed,BN.Flat.MuscHi,BN.Flat.MuscEx,BN.Flat.Vig,BN.Flat.VigEx,BN.Slim.Del,BN.Slim.Norm,BN.Slim.Smo,BN.Slim.Fat,BN.Slim.MuscNo,BN.Slim.MuscLes,BN.Slim.MuscLow,BN.Slim.MuscMed,BN.Slim.MuscHi,BN.Slim.MuscEx,BN.Slim.Vig,BN.Slim.VigEx]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
						10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HaN.Vig at position 10 at random.
					
					Trimming remaining subgroups within Leyenda Skin 4.0 by the required/excluded subgroups of HaN.Vig
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM , BN.Crvy.Del , BS , BSP.Slk , HaD.Def , HaN.Vig
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Vig,BN.Flat.Vig,BN.Slim.Vig]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
						11: [HaS]
					12: [HaSP.Def]
					
					Chose next subgroup: HaS at position 11 at random.
					
					Current Combination: HD.Def.Def , HN.Def , HS , HSP.Def , HDE.BDM , BD.Flat.Def.BN.LStripM , BN.Crvy.Del , BS , BSP.Slk , HaD.Def , HaN.Vig , HaS
					
					Available Subgroups:
					0: [HD.Def.Def,HD.Def.DefM]
					1: [HN.Def]
					2: [HS]
					3: [HSP.Def]
					4: [HDE.BDM]
					5: [BD.Flat.Def.BN.LStripM]
					6: [BN.Crvy.Vig,BN.Flat.Vig,BN.Slim.Vig]
					7: [BS]
					8: [BSP.Gb,BSP.Slk]
					9: [HaD.Def,HaD.Mol]
					10: [HaN.Def,HaN.Vig,HaN.VigEx]
					11: [HaS]
						12: [HaSP.Def]
					
					Chose next subgroup: HaSP.Def at position 12 at random.
					
					Successfully generated combination: Leyenda Skin 4.0:HD.Def.Def|HN.Def|HS|HSP.Def|HDE.BDM|BD.Flat.Def.BN.LStripM|BN.Crvy.Del|BS|BSP.Slk|HaD.Def|HaN.Vig|HaS|HaSP.Def
```

**Comment by Piranha91** [comment]
You're right, the patcher was only checking required/excluded subgroups for the "Available Subgroups" list, but not for the "Current Combination". Pretty significant bug; thank you very much for the catch. I have a fix on my end, just need to figure out why I'm having trouble writing the output .esp after updating the Mutagen backend; after that I can upload a fixed release.

**Comment by Piranha91** [comment]
Should be fixed now in 0.9.7.

**Comment by leonardovcr** [comment]
So, now I'm having trouble with the new version (standalone)... it has been 30 minutes since I started it but it's still loading. Did something change which could lead to this behaviour?

**Comment by Piranha91** [comment]
I updated the Mutagen backend but it shouldn’t cause anything like that, and I’m not getting long loads. The one thing I can think of is I very slightly changed how the patcher decides if a bodyslide xml already exists in your OBody settings or if it needs to be imported. I don’t think it should cause long load times but maybe try a profile with your bodyslide mods disabled? Or even backup and delete the OBody settings json file?

**Comment by leonardovcr** [comment]
I managed to launch after usual loading time (around 10 minutes) after disabling all my bodyslide mods and deleting the obodysettings.json... that probably needs some optimization :)

**Comment by Piranha91** [comment]
It usually takes you 10 min to launch? How many bodyslides do you have?

**Comment by leonardovcr** [comment]
After closing and reopening, I didn't see any big difference in loading time... actually it was a little bit quicker.

I think that including male, female and those included variations like outfit bodyslides, there might be around... 400? Some are unused (like outfits) but I had to clone lots of them to add descriptors for varying weights (extremely boring job), so the real number in the .json file might be around two times that 🤔 

I also know of another user who's having even longer loading times than I do (Rad4r post here: https://www.nexusmods.com/skyrimspecialedition/mods/73717?tab=posts), on a setup with multiple active asset packs. Maybe it's not only bodyslide numbers that are degrading performance?!

If I may suggest, an initialisation log would come in handy in understading why it takes so much time to load on heavy setups...

**Comment by Piranha91** [comment]
Not a bad idea. I need to do a pass at LINQ optimization and I’ll include an init log. I don’t know if it’ll ever be super snappy but it shouldn’t be this slow.

**Comment by leonardovcr** [comment]
Just an update, still trying to load latest version with all bodyslides/config files. It has been hours and still nothing. I've noticed that RAM usage is really abnormal 😕 

![image](https://user-images.githubusercontent.com/13605450/222469018-13e72c9f-8351-4a39-8263-3cf613e32a66.png)

12GB is more than all my asset packs combined 😄

**Comment by Piranha91** [comment]
I added a startup log in 0.9.7.1 - can you try to run it and send it to me? Based on my initial findings it looks like asset packs may be the heavy hitters, but I'd like to corroborate that. 

Btw, I opened a new issue for this: https://github.com/Synthesis-Collective/SynthEBD/issues/106. Let's move this discussion there and close this issue if your subgroups are now getting paired correctly.

**Comment by leonardovcr** [comment]
Fixed now on 0.9.7.1, thanks Piranha 👍
